sklearn/neighbors/tests/test_ball_tree.py

```diff
@@ -65,3 +73,14 @@ def test_query_haversine():
 
     assert_array_almost_equal(dist1, dist2)
     assert_array_almost_equal(ind1, ind2)
+
+
+def test_array_object_type():
```

Comment:
I did not think about the first time, but could you test BallTree and KDTree with a `pytest.mark.parametrize`

Reply:
That was my idea at the beginning, but there is a [test_kd_tree.py](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neighbors/tests/test_kd_tree.py)... with no test inside... and no other tests are parametrized in the same way... why there is no specific test for KDTree?

Reply:
Uhm interesting :). You can add duplicate the test in the file there for the moment.
We should see later on for a refactoring.

Reply:
I don't like the idea of mixing tests for `KDTree` and `BallTree` when a `test_kd_tree.py` is available in the distribution. I find it confusing for people reading the code for the first time (like me? :) ). Let me add the test for `KDTree` in the `test_kd_tree.py` file, in this PR. I will open an issue where refactoring will be discussed after the release. Is that an acceptable solution?

Reply:
Yes it was what I proposed in my previous comment (duplicate the test in `test_kd_tree.py`).

Reply:
Oh, well we agreed then! :)